packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -0,0 +1,218 @@
+
+  stateChanged(state: State) {
+    this.gltfUrl = getGltfUrl(state);
+    this.config = getConfig(state);
+    this.arConfig = getArConfig(state);
+    this.hotspots = getHotspots(state);
+    this.camera = getCamera(state);
+    this.gltf = getGltfModel(state);
+  }
+
+  getSrcPipeUrl(srcType: string): string {
+    return `https://ppng.io/modelviewereditor-srcs-${srcType}-${this.pipeId}`;
+  }
+
+  // TODO: https://javascript.info/fetch-progress
+  async waitForModel() {
+    await fetch(this.getSrcPipeUrl('gltf'))
+        .then(response => response.blob())
+        .then(blob => {
+          const modelUrl = URL.createObjectURL(blob);
+          this.localModelUrl = modelUrl;
+          reduxStore.dispatch(dispatchGltfUrl(modelUrl));
+          reduxStore.dispatch(dispatchSetHotspots([]));
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+  }
+
+  async waitForState(envChanged: boolean) {
+    let partialState: any = {};
+    await fetch(this.snippetPipeUrl)
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error('Something went wrong');
+          }
+        })
+        .then((responseJson) => {
+          partialState = responseJson;
+        })
+        .catch((error) => {
+          console.log('error', error);
+        });
+
+    // These links would be corresponding to the original editor's link.
+    if (envChanged) {
+      partialState.config.environmentImage = undefined;
+    } else if (this.config.environmentImage) {
+      partialState.config.environmentImage = this.config.environmentImage;
+    }
+    partialState.config.src = this.gltfUrl;
+
+    reduxStore.dispatch(dispatchSetHotspots(partialState.hotspots));
+    reduxStore.dispatch(dispatchSetCamera(partialState.camera));
+    reduxStore.dispatch(dispatchSetConfig(partialState.config));
+    reduxStore.dispatch(dispatchArConfig(partialState.arConfig));
+  }
+
+  async waitForEnv(envIsHdr: boolean) {
+    await fetch(this.getSrcPipeUrl('env'))
+        .then(response => response.blob())
+        .then(blob => {
+          // simulating createBlobUrlFromEnvironmentImage
+          const addOn = envIsHdr ? '#.hdr' : '';
+          const envUrl = URL.createObjectURL(blob) + addOn;
+          reduxStore.dispatch(dispatchEnvrionmentImage(envUrl));
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+  }
+
+  async waitForData(json: any) {
+    if (json.gltfChanged) {
+      await this.waitForModel();
+    }
+    if (json.stateChanged) {
+      await this.waitForState(json.envChanged);
+    }
+    if (json.envChanged) {
+      await this.waitForEnv(json.envIsHdr);
+    }
+  }
+
+  async fetchLoop() {
+    await fetch(this.updatesPipeUrl)
+        .then(response => response.json())
+        .then(json => this.waitForData(json))
```

Comment:
json is the initial object sent that has properties that determine what has changed: i.e.: 
{gltfChanged: boolean, stateChanged: boolean, envChanged: boolean, envIsHdr: boolean}
and is how we dynamically know what to load

Reply:
Generally you either use `.then` or `await` but not both, since they do the same thing. Let's stick with `await`.